homeassistant/components/legrand_rflc/light.py

```diff
@@ -0,0 +1,180 @@
+"""The Legrand RFLC integration light platform."""
+
+from collections.abc import Mapping
+from typing import Final
+
+from homeassistant import config_entries
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_TRANSITION,
+    COLOR_MODE_BRIGHTNESS,
+    SUPPORT_TRANSITION,
+    LightEntity,
+)
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+
+class _LegrandRflcSwitch(LightEntity):
+    _attr_should_poll = False
+
+    def __init__(self, hub, zid: int, properties: Mapping):
+        self._hub = hub
+        self._attr_name = properties[hub.NAME]
+        self._attr_is_on = properties[hub.POWER]
+        self._zid = zid
+        hub.on(
+            f"{hub.EVENT_ZONE_PROPERTIES_CHANGED}:{zid}",
+            self._zone_properties_changed,
+        )
+        hub.on(hub.EVENT_CONNECTED, self._available)
+        hub.on(hub.EVENT_DISCONNECTED, self._available)
+        hub.on(hub.EVENT_AUTHENTICATED, self._available)
+        hub.on(hub.EVENT_UNAUTHENTICATED, self._available)
+
+    @property
+    def unique_id(self) -> str:
+        return f"{self._hub.host()}:{self._zid}"
+
+    @property
+    def available(self) -> bool:
+        return self._hub.connected and self._hub.authenticated
+
+    async def _available(self) -> None:
+        self.async_write_ha_state()
+
+    def _zone_properties_changed_switch(self, message: Mapping) -> None:
+        hub = self._hub
+        if hub.PROPERTY_LIST in message:
+            properties = message[hub.PROPERTY_LIST]
+            if hub.POWER in properties:
+                self._attr_is_on = properties[hub.POWER]
+
+    async def _zone_properties_changed(self, message: Mapping) -> None:
+        self._zone_properties_changed_switch(message)
+        self.async_write_ha_state()
+
+    async def _async_switch(self, power: bool) -> None:
+        hub = self._hub
+
+        async def handle(message: Mapping):
```

Comment:
Looks like this could be a method so we don't need to define it every service call.

Reply:
https://github.com/home-assistant/core/pull/53430/commits/8a45a1f3a49a95ee186085d3251be946097d12e8